### ISSUE TITLE
Can not build CRAXplusplus in ubuntu20.04

### ISSUE THREAD

**Issue by LJP-TW** [SQLab/CRAXplusplus#11]
I tried to build this project following the [documentation](https://github.com/SQLab/CRAXplusplus/blob/881c9f3f43f1e406bc2f96f50f72ee665516d249/Documentation/Build.md) but got the following error (This is just part of the log):
```
...

make -C libs2e
make[3]: Entering directory '/home/pt/s2e/build/libs2e-release/i386-softmmu/libs2e'                                                                         make[3]: *** No targets specified and no makefile found.  Stop.
make[3]: Leaving directory '/home/pt/s2e/build/libs2e-release/i386-softmmu/libs2e'                                                                          make[2]: *** [Makefile:59: libs2e/src/libs2e.so] Error 2
make[2]: Leaving directory '/home/pt/s2e/build/libs2e-release/i386-softmmu'
make[1]: *** [Makefile:27: i386-softmmu] Error 2

...

make -C libs2e
make[3]: Entering directory '/home/pt/s2e/build/libs2e-release/x86_64-softmmu/libs2e'
make[3]: *** No targets specified and no makefile found.  Stop.
make[3]: Leaving directory '/home/pt/s2e/build/libs2e-release/x86_64-softmmu/libs2e'
make[2]: *** [Makefile:59: libs2e/src/libs2e.so] Error 2
make[2]: Leaving directory '/home/pt/s2e/build/libs2e-release/x86_64-softmmu' make[1]: *** [Makefile:27: x86_64-softmmu] Error 2

...

make -C libs2eplugins
make[3]: Entering directory '/home/pt/s2e/build/libs2e-release/x86_64-s2e-softmmu/libs2eplugins'                                                            make[3]: *** No targets specified and no makefile found.  Stop.
make[3]: Leaving directory '/home/pt/s2e/build/libs2e-release/x86_64-s2e-softmmu/libs2eplugins'
make[2]: *** [Makefile:51: libs2eplugins/src/libs2eplugins.a] Error 2
make[2]: Leaving directory '/home/pt/s2e/build/libs2e-release/x86_64-s2e-softmmu'
make[1]: *** [Makefile:27: x86_64-s2e-softmmu] Error 2

...

make -C libs2eplugins
make[3]: Entering directory '/home/pt/s2e/build/libs2e-release/i386-s2e_sp-softmmu/libs2eplugins'
make[3]: *** No targets specified and no makefile found.  Stop.
make[3]: Leaving directory '/home/pt/s2e/build/libs2e-release/i386-s2e_sp-softmmu/libs2eplugins'                                                            make[2]: *** [Makefile:51: libs2eplugins/src/libs2eplugins.a] Error 2
make[2]: Leaving directory '/home/pt/s2e/build/libs2e-release/i386-s2e_sp-softmmu'
make[1]: *** [Makefile:27: i386-s2e_sp-softmmu] Error 2

...

make -C libs2eplugins
make[3]: Entering directory '/home/pt/s2e/build/libs2e-release/x86_64-s2e_sp-softmmu/libs2eplugins'
make[3]: *** No targets specified and no makefile found.  Stop.
make[3]: Leaving directory '/home/pt/s2e/build/libs2e-release/x86_64-s2e_sp-softmmu/libs2eplugins'
make[2]: *** [Makefile:51: libs2eplugins/src/libs2eplugins.a] Error 2
make[2]: Leaving directory '/home/pt/s2e/build/libs2e-release/x86_64-s2e_sp-softmmu'
make[1]: *** [Makefile:27: x86_64-s2e_sp-softmmu] Error 2
make[1]: Leaving directory '/home/pt/s2e/build/libs2e-release'
make: *** [/home/pt/s2e/source/scripts//..//s2e/Makefile:210: stamps/libs2e-release-make] Error 2
make: Leaving directory '/home/pt/s2e/build'
ERROR: [build]

  RAN: /usr/bin/make --directory=/home/pt/s2e/build --file=/home/pt/s2e/source/Makefile install

  STDOUT:


  STDERR:
```

I think the problem is that cmake is not working properly. The following log appears before the above error log:
```
CMake Error at CMakeLists.txt:92 (find_package):
  By not providing "Findpybind11.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "pybind11",
  but CMake did not find one.

  Could not find a package configuration file provided by "pybind11" with any
  of the following names:                                                     
    pybind11Config.cmake
    pybind11-config.cmake

  Add the installation prefix of "pybind11" to CMAKE_PREFIX_PATH or set
  "pybind11_DIR" to a directory containing one of the above files.  If
  "pybind11" provides a separate development package or SDK, be sure it has
  been installed.


Failed to configure libs2e
CMake Error at CMakeLists.txt:92 (find_package):
  By not providing "Findpybind11.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "pybind11",
  but CMake did not find one.

  Could not find a package configuration file provided by "pybind11" with any
  of the following names:

    pybind11Config.cmake
    pybind11-config.cmake

  Add the installation prefix of "pybind11" to CMAKE_PREFIX_PATH or set
  "pybind11_DIR" to a directory containing one of the above files.  If
  "pybind11" provides a separate development package or SDK, be sure it has
  been installed.


Failed to configure libs2e
CMake Error at CMakeLists.txt:106 (find_package):
  By not providing "Findpybind11.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "pybind11",
  but CMake did not find one.

  Could not find a package configuration file provided by "pybind11" with any
  of the following names:

    pybind11Config.cmake
    pybind11-config.cmake

  Add the installation prefix of "pybind11" to CMAKE_PREFIX_PATH or set
  "pybind11_DIR" to a directory containing one of the above files.  If
  "pybind11" provides a separate development package or SDK, be sure it has
  been installed.


Failed to configure libs2eplugins
CMake Error at CMakeLists.txt:106 (find_package):
  By not providing "Findpybind11.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "pybind11",
  but CMake did not find one.

  Could not find a package configuration file provided by "pybind11" with any
  of the following names:

    pybind11Config.cmake
    pybind11-config.cmake

  Add the installation prefix of "pybind11" to CMAKE_PREFIX_PATH or set
  "pybind11_DIR" to a directory containing one of the above files.  If
  "pybind11" provides a separate development package or SDK, be sure it has
  been installed.


Failed to configure libs2eplugins
CMake Error at CMakeLists.txt:106 (find_package):
  By not providing "Findpybind11.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "pybind11",
  but CMake did not find one.

  Could not find a package configuration file provided by "pybind11" with any
  of the following names:

    pybind11Config.cmake
    pybind11-config.cmake

  Add the installation prefix of "pybind11" to CMAKE_PREFIX_PATH or set
  "pybind11_DIR" to a directory containing one of the above files.  If
  "pybind11" provides a separate development package or SDK, be sure it has
  been installed.


Failed to configure libs2eplugins
CMake Error at CMakeLists.txt:106 (find_package):
  By not providing "Findpybind11.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "pybind11",
  but CMake did not find one.

  Could not find a package configuration file provided by "pybind11" with any
  of the following names:

    pybind11Config.cmake
    pybind11-config.cmake

  Add the installation prefix of "pybind11" to CMAKE_PREFIX_PATH or set
  "pybind11_DIR" to a directory containing one of the above files.  If
  "pybind11" provides a separate development package or SDK, be sure it has
  been installed.


Failed to configure libs2eplugins
```

The first time I build this project isn't in home directory. But this time I build this in home directory and the error message is the same as before.

**Comment by aesophor** [comment]
Could you please check the content of `/usr/include/pybind11` on your build machine?

I'll look into this issue in detail after I get off my work today.

```
/home/aesophor [aesophor@aesophor-vm] [15:33]
> ll /usr/include/pybind11/
total 472K
drwxr-xr-x 2 root root 4.0K Nov 21  2021 detail
-rw-r--r-- 1 root root  19K Oct 15  2019 attr.h
-rw-r--r-- 1 root root 4.3K Oct 15  2019 buffer_info.h
-rw-r--r-- 1 root root  88K Oct 15  2019 cast.h
-rw-r--r-- 1 root root 7.6K Oct 15  2019 chrono.h
-rw-r--r-- 1 root root  120 Oct 15  2019 common.h
-rw-r--r-- 1 root root 2.0K Oct 15  2019 complex.h
-rw-r--r-- 1 root root  29K Oct 15  2019 eigen.h
-rw-r--r-- 1 root root 7.6K Oct 15  2019 embed.h
-rw-r--r-- 1 root root 3.8K Oct 15  2019 eval.h
-rw-r--r-- 1 root root 3.6K Oct 15  2019 functional.h
-rw-r--r-- 1 root root 5.6K Oct 15  2019 iostream.h
-rw-r--r-- 1 root root  67K Oct 15  2019 numpy.h
-rw-r--r-- 1 root root 8.6K Oct 15  2019 operators.h
-rw-r--r-- 1 root root 2.0K Oct 15  2019 options.h
-rw-r--r-- 1 root root  96K Oct 15  2019 pybind11.h
-rw-r--r-- 1 root root  57K Oct 15  2019 pytypes.h
-rw-r--r-- 1 root root  23K Oct 15  2019 stl_bind.h
-rw-r--r-- 1 root root  14K Oct 15  2019 stl.h
```

**Comment by LJP-TW** [comment]
Oh I didn't have pybind11 installed, my bad:(

I build successfully after using the following command:
```
sudo apt-get install pybind11-dev=2.4.3-2build2
# Rebuild S2E
rm -rf ~/s2e/build/*
s2e build
```

But when I run CRAX++ I have the following problem:
```
# ./launch-crax.sh
Starting libs2e...
Opening /dev/kvm
Initializing qemu64-s2e cpu
Using module /home/pt/s2e/install/share/libs2e/op_helper.bc.x86_64
S2E: output directory = "./s2e-out-0"
KLEE: WARNING: unsupported intrinsic llvm.rint.f64
KLEE: WARNING: unsupported intrinsic llvm.fmuladd.f64
Using log level override 'info'
Setting console level to 'info'
Creating plugin CorePlugin
Creating plugin BaseInstructions
Creating plugin HostFiles
Creating plugin Vmi
Creating plugin MemUtils
Creating plugin WebServiceInterface
Creating plugin ExecutionTracer
Creating plugin ModuleTracer
Creating plugin KeyValueStore
Creating plugin TranslationBlockCoverage
Creating plugin ModuleExecutionDetector
Creating plugin ForkLimiter
Creating plugin ProcessExecutionDetector
Creating plugin ModuleMap
Creating plugin MemoryMap
Creating plugin MultiSearcher
Creating plugin CUPASearcher
Creating plugin FunctionModels
Creating plugin LinuxMonitor
Creating plugin LuaBindings
Creating plugin LuaCoreEvents
Creating plugin CRAX
/home/pt/s2e/install/bin/qemu-system-x86_64: symbol lookup error: /home/pt/s2e/install/share/libs2e/libs2e-x86_64-s2e.so: undefined symbol: _ZN3s2e7plugins4crax5ProxyC1Ev
```

The symbol name is `s2e::plugins::crax::Proxy::Proxy()`. And I think the code of this symbol is in `libs2eplugins.a`

The `~/s2e/install/share/libs2e/libs2e-x86_64-s2e.so` is copy from `~/s2e/build/libs2e-release/x86_64-s2e-softmmu/libs2e.so`

So I check the compiling command for this file:
```
[100%] Linking CXX shared library libs2e.so
cd /home/pt/s2e/build/libs2e-release/x86_64-s2e-softmmu/libs2e/src && /usr/bin/cmake -E cmake_link_script CMakeFiles/s2e.dir/link.txt --verbose=1
/home/pt/s2e/install/bin/clang++ -fPIC -march=native -Werror -Wno-zero-length-array -Wno-c99-extensions                            -Wno-gnu-anonymous-struct -Wno-nested-anon-types                              -Wno-gnu-statement-expression                                                 -Wno-gnu-zero-variadic-macro-arguments -Wno-vla-extension                     -Wno-covered-switch-default -Wno-shift-negative-value                         -Wno-deprecated-register -Wno-sign-compare                                    -Wno-missing-field-initializers -Wno-mismatched-tags                          -Wno-deprecated-declarations -Wno-initializer-overrides                       -Wno-zero-length-array -D__STDC_FORMAT_MACROS -D_GNU_SOURCE -DNEED_CPU_H  -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -DTARGET_PHYS_ADDR_BITS=64 -Wall -fPIC -fno-strict-aliasing -fexceptions -std=c++14 -O2 -g  -Wl,--version-script=/home/pt/s2e/source/s2e/libs2e/src/mapfile -shared -Wl,-soname,libs2e.so -o libs2e.so CMakeFiles/s2e.dir/libs2e.cpp.o CMakeFiles/s2e.dir/s2e-kvm-io.cpp.o CMakeFiles/s2e.dir/s2e-kvm-state.cpp.o CMakeFiles/s2e.dir/s2e-kvm-trace.cpp.o CMakeFiles/s2e.dir/s2e-kvm.cpp.o CMakeFiles/s2e.dir/s2e-kvm-vm.cpp.o CMakeFiles/s2e.dir/s2e-kvm-vcpu.cpp.o CMakeFiles/s2e.dir/s2e-libcpu-interface.cpp.o CMakeFiles/s2e.dir/crashdump.cpp.o   -L/home/pt/s2e/build/libs2e-release/x86_64-s2e-softmmu/libtcg/src  -L/home/pt/s2e/build/libs2e-release/x86_64-s2e-softmmu/libcpu/src  -L/home/pt/s2e/build/libq-release/src  -L/home/pt/s2e/build/klee-release/lib  -L/home/pt/s2e/build/lua-5.3.4/src  -L/home/pt/s2e/build/libfsigc++-release/src  -L/home/pt/s2e/build/libcoroutine-release/src  -L/home/pt/s2e/build/libs2e-release/x86_64-s2e-softmmu/libs2ecore/src  -L/home/pt/s2e/build/libs2e-release/x86_64-s2e-softmmu/libs2eplugins/src  -L/home/pt/s2e/build/libvmi-release/src  -Wl,-rpath,/home/pt/s2e/build/libs2e-release/x86_64-s2e-softmmu/libtcg/src:/home/pt/s2e/build/libs2e-release/x86_64-s2e-softmmu/libcpu/src:/home/pt/s2e/build/libq-release/src:/home/pt/s2e/build/klee-release/lib:/home/pt/s2e/build/lua-5.3.4/src:/home/pt/s2e/build/libfsigc++-release/src:/home/pt/s2e/build/libcoroutine-release/src:/home/pt/s2e/build/libs2e-release/x86_64-s2e-softmmu/libs2ecore/src:/home/pt/s2e/build/libs2e-release/x86_64-s2e-softmmu/libs2eplugins/src:/home/pt/s2e/build/libvmi-release/src:/home/pt/s2e/build/llvm-release/lib -Wl,--whole-archive /home/pt/s2e/build/libs2e-release/x86_64-s2e-softmmu/libs2eplugins/src/libs2eplugins.a /home/pt/s2e/build/libs2e-release/x86_64-s2e-softmmu/libs2ecore/src/libs2ecore.a -Wl,--no-whole-archive /home/pt/s2e/build/libs2e-release/x86_64-s2e-softmmu/libcpu/src/libcpu.a /home/pt/s2e/build/libs2e-release/x86_64-s2e-softmmu/libtcg/src/libtcg.a /home/pt/s2e/build/klee-release/lib/libkleeCore.a /home/pt/s2e/build/klee-release/lib/libkleeModule.a /home/pt/s2e/build/klee-release/lib/libkleaverSolver.a /home/pt/s2e/build/klee-release/lib/libkleaverExpr.a /home/pt/s2e/build/klee-release/lib/libkleeSupport.a /home/pt/s2e/build/klee-release/lib/libkleeBasic.a /home/pt/s2e/build/libvmi-release/src/libvmi.a -lelf -lmemcached -llua /home/pt/s2e/build/llvm-release/lib/libLLVMDemangle.a /home/pt/s2e/build/llvm-release/lib/libLLVMSupport.a /home/pt/s2e/build/llvm-release/lib/libLLVMTableGen.a /home/pt/s2e/build/llvm-release/lib/libLLVMTableGenGlobalISel.a /home/pt/s2e/build/llvm-release/lib/libLLVMCore.a /home/pt/s2e/build/llvm-release/lib/libLLVMFuzzMutate.a /home/pt/s2e/build/llvm-release/lib/libLLVMFileCheck.a /home/pt/s2e/build/llvm-release/lib/libLLVMInterfaceStub.a /home/pt/s2e/build/llvm-release/lib/libLLVMIRReader.a /home/pt/s2e/build/llvm-release/lib/libLLVMCodeGen.a /home/pt/s2e/build/llvm-release/lib/libLLVMSelectionDAG.a /home/pt/s2e/build/llvm-release/lib/libLLVMAsmPrinter.a /home/pt/s2e/build/llvm-release/lib/libLLVMMIRParser.a /home/pt/s2e/build/llvm-release/lib/libLLVMGlobalISel.a /home/pt/s2e/build/llvm-release/lib/libLLVMBinaryFormat.a /home/pt/s2e/build/llvm-release/lib/libLLVMBitReader.a /home/pt/s2e/build/llvm-release/lib/libLLVMBitWriter.a /home/pt/s2e/build/llvm-release/lib/libLLVMBitstreamReader.a /home/pt/s2e/build/llvm-release/lib/libLLVMDWARFLinker.a /home/pt/s2e/build/llvm-release/lib/libLLVMExtensions.a /home/pt/s2e/build/llvm-release/lib/libLLVMFrontendOpenACC.a /home/pt/s2e/build/llvm-release/lib/libLLVMFrontendOpenMP.a /home/pt/s2e/build/llvm-release/lib/libLLVMTransformUtils.a /home/pt/s2e/build/llvm-release/lib/libLLVMInstrumentation.a /home/pt/s2e/build/llvm-release/lib/libLLVMAggressiveInstCombine.a /home/pt/s2e/build/llvm-release/lib/libLLVMInstCombine.a /home/pt/s2e/build/llvm-release/lib/libLLVMScalarOpts.a /home/pt/s2e/build/llvm-release/lib/libLLVMipo.a /home/pt/s2e/build/llvm-release/lib/libLLVMVectorize.a /home/pt/s2e/build/llvm-release/lib/libLLVMObjCARCOpts.a /home/pt/s2e/build/llvm-release/lib/libLLVMCoroutines.a /home/pt/s2e/build/llvm-release/lib/libLLVMCFGuard.a /home/pt/s2e/build/llvm-release/lib/libLLVMLinker.a /home/pt/s2e/build/llvm-release/lib/libLLVMAnalysis.a /home/pt/s2e/build/llvm-release/lib/libLLVMLTO.a /home/pt/s2e/build/llvm-release/lib/libLLVMMC.a /home/pt/s2e/build/llvm-release/lib/libLLVMMCParser.a /home/pt/s2e/build/llvm-release/lib/libLLVMMCDisassembler.a /home/pt/s2e/build/llvm-release/lib/libLLVMMCA.a /home/pt/s2e/build/llvm-release/lib/libLLVMObject.a /home/pt/s2e/build/llvm-release/lib/libLLVMObjectYAML.a /home/pt/s2e/build/llvm-release/lib/libLLVMOption.a /home/pt/s2e/build/llvm-release/lib/libLLVMRemarks.a /home/pt/s2e/build/llvm-release/lib/libLLVMDebuginfod.a /home/pt/s2e/build/llvm-release/lib/libLLVMDebugInfoDWARF.a /home/pt/s2e/build/llvm-release/lib/libLLVMDebugInfoGSYM.a /home/pt/s2e/build/llvm-release/lib/libLLVMDebugInfoMSF.a /home/pt/s2e/build/llvm-release/lib/libLLVMDebugInfoCodeView.a /home/pt/s2e/build/llvm-release/lib/libLLVMDebugInfoPDB.a /home/pt/s2e/build/llvm-release/lib/libLLVMSymbolize.a /home/pt/s2e/build/llvm-release/lib/libLLVMDWP.a /home/pt/s2e/build/llvm-release/lib/libLLVMExecutionEngine.a /home/pt/s2e/build/llvm-release/lib/libLLVMInterpreter.a /home/pt/s2e/build/llvm-release/lib/libLLVMJITLink.a /home/pt/s2e/build/llvm-release/lib/libLLVMMCJIT.a /home/pt/s2e/build/llvm-release/lib/libLLVMOrcJIT.a /home/pt/s2e/build/llvm-release/lib/libLLVMOrcShared.a /home/pt/s2e/build/llvm-release/lib/libLLVMOrcTargetProcess.a /home/pt/s2e/build/llvm-release/lib/libLLVMRuntimeDyld.a /home/pt/s2e/build/llvm-release/lib/libLLVMTarget.a /home/pt/s2e/build/llvm-release/lib/libLLVMX86CodeGen.a /home/pt/s2e/build/llvm-release/lib/libLLVMX86AsmParser.a /home/pt/s2e/build/llvm-release/lib/libLLVMX86Disassembler.a /home/pt/s2e/build/llvm-release/lib/libLLVMX86TargetMCA.a /home/pt/s2e/build/llvm-release/lib/libLLVMX86Desc.a /home/pt/s2e/build/llvm-release/lib/libLLVMX86Info.a /home/pt/s2e/build/llvm-release/lib/libLLVMAsmParser.a /home/pt/s2e/build/llvm-release/lib/libLLVMLineEditor.a /home/pt/s2e/build/llvm-release/lib/libLLVMProfileData.a /home/pt/s2e/build/llvm-release/lib/libLLVMCoverage.a /home/pt/s2e/build/llvm-release/lib/libLLVMPasses.a /home/pt/s2e/build/llvm-release/lib/libLLVMTextAPI.a /home/pt/s2e/build/llvm-release/lib/libLLVMDlltoolDriver.a /home/pt/s2e/build/llvm-release/lib/libLLVMLibDriver.a /home/pt/s2e/build/llvm-release/lib/libLLVMXRay.a /home/pt/s2e/build/llvm-release/lib/libLLVMWindowsManifest.a /home/pt/s2e/build/llvm-release/lib/libLTO.so.14 /home/pt/s2e/build/llvm-release/lib/libLLVMCFIVerify.a /home/pt/s2e/build/llvm-release/lib/libLLVMDiff.a /home/pt/s2e/build/llvm-release/lib/libLLVMExegesisX86.a /home/pt/s2e/build/llvm-release/lib/libLLVMExegesis.a /home/pt/s2e/build/llvm-release/lib/libRemarks.so.14 -Wl,-Bstatic -lz3 -Wl,-Bdynamic -lgomp /home/pt/s2e/build/libq-release/src/libq.a /home/pt/s2e/install/lib/libprotobuf.a -lpthread -lboost_serialization -lboost_system -lboost_regex /home/pt/s2e/build/libcoroutine-release/src/libcoroutine.a /home/pt/s2e/build/libfsigc++-release/src/libfsigc++.a /home/pt/s2e/install/lib/libcapstone.a -lpthread -lglib-2.0 -lbsd /home/pt/s2e/build/llvm-release/lib/libLLVMTableGen.a /home/pt/s2e/build/llvm-release/lib/libLLVMExtensions.a /home/pt/s2e/build/llvm-release/lib/libLLVMJITLink.a /home/pt/s2e/build/llvm-release/lib/libLLVMObjCARCOpts.a /home/pt/s2e/build/llvm-release/lib/libLLVMCoroutines.a /home/pt/s2e/build/llvm-release/lib/libLLVMipo.a /home/pt/s2e/build/llvm-release/lib/libLLVMIRReader.a /home/pt/s2e/build/llvm-release/lib/libLLVMAsmParser.a /home/pt/s2e/build/llvm-release/lib/libLLVMFrontendOpenMP.a /home/pt/s2e/build/llvm-release/lib/libLLVMVectorize.a /home/pt/s2e/build/llvm-release/lib/libLLVMLinker.a /home/pt/s2e/build/llvm-release/lib/libLLVMMCA.a /home/pt/s2e/build/llvm-release/lib/libLLVMOption.a /home/pt/s2e/build/llvm-release/lib/libLLVMSymbolize.a /home/pt/s2e/build/llvm-release/lib/libLLVMDebugInfoPDB.a /home/pt/s2e/build/llvm-release/lib/libLLVMX86CodeGen.a /home/pt/s2e/build/llvm-release/lib/libLLVMAsmPrinter.a /home/pt/s2e/build/llvm-release/lib/libLLVMDebugInfoMSF.a /home/pt/s2e/build/llvm-release/lib/libLLVMInstrumentation.a /home/pt/s2e/build/llvm-release/lib/libLLVMCFGuard.a /home/pt/s2e/build/llvm-release/lib/libLLVMX86AsmParser.a /home/pt/s2e/build/llvm-release/lib/libLLVMX86Disassembler.a /home/pt/s2e/build/llvm-release/lib/libLLVMX86Desc.a /home/pt/s2e/build/llvm-release/lib/libLLVMX86Info.a /home/pt/s2e/build/llvm-release/lib/libLLVMGlobalISel.a /home/pt/s2e/build/llvm-release/lib/libLLVMSelectionDAG.a /home/pt/s2e/build/llvm-release/lib/libLLVMCodeGen.a /home/pt/s2e/build/llvm-release/lib/libLLVMBitWriter.a /home/pt/s2e/build/llvm-release/lib/libLLVMScalarOpts.a /home/pt/s2e/build/llvm-release/lib/libLLVMAggressiveInstCombine.a /home/pt/s2e/build/llvm-release/lib/libLLVMInstCombine.a /home/pt/s2e/build/llvm-release/lib/libLLVMTransformUtils.a /home/pt/s2e/build/llvm-release/lib/libLLVMMCDisassembler.a /home/pt/s2e/build/llvm-release/lib/libLLVMObjectYAML.a /home/pt/s2e/build/llvm-release/lib/libLLVMMCJIT.a /home/pt/s2e/build/llvm-release/lib/libLLVMExecutionEngine.a /home/pt/s2e/build/llvm-release/lib/libLLVMOrcTargetProcess.a /home/pt/s2e/build/llvm-release/lib/libLLVMOrcShared.a /home/pt/s2e/build/llvm-release/lib/libLLVMTarget.a /home/pt/s2e/build/llvm-release/lib/libLLVMAnalysis.a /home/pt/s2e/build/llvm-release/lib/libLLVMProfileData.a /home/pt/s2e/build/llvm-release/lib/libLLVMDebugInfoDWARF.a /home/pt/s2e/build/llvm-release/lib/libLLVMRuntimeDyld.a /home/pt/s2e/build/llvm-release/lib/libLLVMObject.a /home/pt/s2e/build/llvm-release/lib/libLLVMBitReader.a /home/pt/s2e/build/llvm-release/lib/libLLVMCore.a /home/pt/s2e/build/llvm-release/lib/libLLVMRemarks.a /home/pt/s2e/build/llvm-release/lib/libLLVMBitstreamReader.a /home/pt/s2e/build/llvm-release/lib/libLLVMMCParser.a /home/pt/s2e/build/llvm-release/lib/libLLVMMC.a /home/pt/s2e/build/llvm-release/lib/libLLVMDebugInfoCodeView.a /home/pt/s2e/build/llvm-release/lib/libLLVMTextAPI.a /home/pt/s2e/build/llvm-release/lib/libLLVMBinaryFormat.a /home/pt/s2e/build/llvm-release/lib/libLLVMSupport.a /home/pt/s2e/build/llvm-release/lib/libLLVMDemangle.a -lrt -ldl -lpthread -lm /usr/lib/x86_64-linux-gnu/libpython3.8.so
```

In particular, This part of the log above shows that `libs2e.so` should contain every object in `libs2eplugins.a`:
`-Wl,--whole-archive /home/pt/s2e/build/libs2e-release/x86_64-s2e-softmmu/libs2eplugins/src/libs2eplugins.a`

But it turns out that `libs2eplugins.a` doesn't have the code for `s2e::plugins::crax::Proxy::Proxy()`

Then I check `~/s2e/source/s2e/libs2eplugins/src/CMakeLists.txt`, which describes how `libs2eplugins.a` was compiled.

The `CMakeLists.txt` is actually patched by [libs2eplugins.patch](https://github.com/SQLab/CRAXplusplus/blob/f2334354f9a2711097c8d9a41fec38995143a076/patches/libs2eplugins.patch), but this patch misses some source code files (e.g. `s2e/Plugins/CRAX/Proxy.cpp`), and this is the root cause of this problem.

After adding `s2e/Plugins/CRAX/Proxy.cpp` as the source file of libs2eplugins in the `CMakeLists.txt`, then it work :)

**Comment by aesophor** [comment]
Fixed in 25c5f9777780f80782b838da4cc9850d1f08b994. Sorry for your trouble.
